src/components/BulkEnrollmentModal/index.jsx

```diff
@@ -53,24 +57,60 @@ class BulkEnrollmentModal extends React.Component {
     onClose();
   }
 
+  validateFormData(formData) {
+    const userEmailsKey = EMAIL_ADDRESS_TEXT_FORM_DATA;
+    const emailsCSVKey = EMAIL_ADDRESS_CSV_FORM_DATA;
+
+    /* eslint-disable no-underscore-dangle */
+    // The 'subject' field is not required here.
+    let errors = {
+      _error: [],
+    };
+
+    if (!formData[userEmailsKey] && !formData[emailsCSVKey]) {
+      const message = 'Either user emails or emails csv must be provided.';
+      errors[userEmailsKey] = message;
+      errors[emailsCSVKey] = message;
+      errors._error.push(message);
+    }
+
+    const emailFieldErrors = validateEmailAddressesFields(formData);
+    errors = mergeErrors(errors, emailFieldErrors);
+
+    if (errors._error.length > 0) {
+      throw new SubmissionError(errors);
+    }
+    /* eslint-enable no-underscore-dangle */
+  }
+
+  normalizeEmailValues(value) {
+    return value && value.split(/\r\n|\n/);
+  }
+
   handleModalSubmit(formData) {
     const {
       enterpriseUuid,
       sendBulkEnrollment,
       selectedCourseRunKeys,
     } = this.props;
 
-    const bulkEnrollmentPromises = selectedCourseRunKeys.map((selectedCourseRunKey) => {
-      const options = {
-        course_run_key: selectedCourseRunKey,
-        email: formData['bulk-enrollment-email-addresses'],
-        email_csv: formData['bulk-enrollment-email-csv'],
-      };
-      return sendBulkEnrollment(enterpriseUuid, options);
-    });
+    this.validateFormData(formData);
 
-    return Promise.all(bulkEnrollmentPromises)
+    const options = {
+      course_run_keys: selectedCourseRunKeys,
+      notify: true,
+    };
+    const emails = [];
```

Comment:
Talked offline and this is essentially a non-issue as we won't cause any errors if duplicate learners are requested to be enrolled, but if you'd like I think you can dedup fairly easily here by just replacing the `emails`  array for a set and change `.push` for `.add` below. I've (hopefully) verified that `validateEmailAddresses` can handle sets as well as arrays so you should be ok.

Reply:
used lodash `union` instead of necessarily a set here, but ultimately the change is the same: de-dupe the emails regardless of which field they came out of.